Panic??? Need help

Hi, all!
SQLC docker image:

docker run --rm -v /Users/XXX/Workspace/src/gitlab.com/project/be:/src -w /src kjconroy/sqlc generate
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xcfe8a8]

goroutine 1 [running]:
github.com/kyleconroy/sqlc/internal/compiler.resolveCatalogRefs(0xc0004c0230, 0xc00012a648, 0x1, 0x1, 0xc00094c0f0, 0x5, 0x5, 0xc0009cd830, 0x0, 0xc000b90d30, ...)
	/workspace/internal/compiler/resolve.go:301 +0x16c8
github.com/kyleconroy/sqlc/internal/compiler.(*Compiler).parseQuery(0xc0003cc800, 0x12ecb20, 0xc0008f98e0, 0xc000c95900, 0x4d0, 0x0, 0x0, 0x0, 0x0)
	/workspace/internal/compiler/parse.go:90 +0x565
github.com/kyleconroy/sqlc/internal/compiler.(*Compiler).parseQueries(0xc0003cc800, 0x1000000, 0xc0004c0230, 0xc0005233a0, 0x1)
	/workspace/internal/compiler/compile.go:109 +0x599
github.com/kyleconroy/sqlc/internal/compiler.(*Compiler).ParseQueries(...)
	/workspace/internal/compiler/engine.go:49
github.com/kyleconroy/sqlc/internal/cmd.parse(0xc0000dd600, 0xc00052712c, 0x3, 0xc000526edc, 0x4, 0xc000527176, 0xa, 0xc0005233a0, 0x1, 0x1, ...)
	/workspace/internal/cmd/generate.go:223 +0x145
github.com/kyleconroy/sqlc/internal/cmd.Generate(0x19d2300, 0xc000526edc, 0x4, 0x0, 0x0, 0x12f10e0, 0xc000010020, 0x0, 0x10a0b77, 0x32)
	/workspace/internal/cmd/generate.go:167 +0xce9
github.com/kyleconroy/sqlc/internal/cmd.glob..func3(0x19d23a0, 0x1f2a160, 0x0, 0x0)
	/workspace/internal/cmd/cmd.go:120 +0x17d
github.com/spf13/cobra.(*Command).execute(0x19d23a0, 0x1f2a160, 0x0, 0x0, 0x19d23a0, 0x1f2a160)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc00056c780, 0xc0000ddf10, 0x1, 0x1)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/kyleconroy/sqlc/internal/cmd.Do(0xc00003c050, 0x1, 0x1, 0x12f10a0, 0xc000010010, 0x12f10e0, 0xc000010018, 0x12f10e0, 0xc000010020, 0xc000096058)
	/workspace/internal/cmd/cmd.go:34 +0x317
main.main()
	/workspace/cmd/sqlc/main.go:10 +0xb5
make: *** [generate-dbs] Error 2

--- a/internal/compiler/parse.go
+++ b/internal/compiler/parse.go
@@ -87,12 +87,11 @@ func (c *Compiler) parseQuery(stmt ast.Node, src string, o opts.Parser) (*Query,
 		refs = uniqueParamRefs(refs)
 		sort.Slice(refs, func(i, j int) bool { return refs[i].ref.Number < refs[j].ref.Number })
 	}
-	params, err := resolveCatalogRefs(c.catalog, rvs, refs, namedParams)
+	qc, err := buildQueryCatalog(c.catalog, raw.Stmt)
 	if err != nil {
 		return nil, err
 	}
-
-	qc, err := buildQueryCatalog(c.catalog, raw.Stmt)
+	params, err := resolveCatalogRefs(c.catalog, qc, rvs, refs, namedParams)
 	if err != nil {
 		return nil, err
 	}

--- a/internal/compiler/resolve.go
+++ b/internal/compiler/resolve.go
@@ -18,7 +18,7 @@ func dataType(n *ast.TypeName) string {
 	}
 }
 
-func resolveCatalogRefs(c *catalog.Catalog, rvs []*ast.RangeVar, args []paramRef, names map[int]string) ([]Parameter, error) {
+func resolveCatalogRefs(c *catalog.Catalog, qc *QueryCatalog, rvs []*ast.RangeVar, args []paramRef, names map[int]string) ([]Parameter, error) {
 	aliasMap := map[string]*ast.TableName{}
 	// TODO: Deprecate defaultTable
 	var defaultTable *ast.TableName
@@ -62,6 +62,10 @@ func resolveCatalogRefs(c *catalog.Catalog, rvs []*ast.RangeVar, args []paramRef
 		}
 		table, err := c.GetTable(fqn)
 		if err != nil {
+			// If the table name doesn't exist, fisrt check if it's a CTE
+			if _, qcerr := qc.GetTable(fqn); qcerr != nil {
+				return nil, err
+			}
 			continue
 		}
 		err = indexTable(table)
@@ -299,15 +303,19 @@ func resolveCatalogRefs(c *catalog.Catalog, rvs []*ast.RangeVar, args []paramRef
 			if err != nil {
 				return nil, err
 			}
+
 		case *ast.ResTarget:
 			if n.Name == nil {
 				return nil, fmt.Errorf("*ast.ResTarget has nil name")
 			}
 			key := *n.Name
 
+			var schema, rel string
 			// TODO: Deprecate defaultTable
-			schema := defaultTable.Schema
-			rel := defaultTable.Name
+			if defaultTable != nil {
+				schema = defaultTable.Schema
+				rel = defaultTable.Name
+			}
 			if ref.rv != nil {
 				fqn, err := ParseTableName(ref.rv)
 				if err != nil {
@@ -319,7 +327,13 @@ func resolveCatalogRefs(c *catalog.Catalog, rvs []*ast.RangeVar, args []paramRef
 			if schema == "" {
 				schema = c.DefaultSchema
 			}
-			if c, ok := typeMap[schema][rel][key]; ok {
+
+			tableMap, ok := typeMap[schema][rel]
+			if !ok {
+				return nil, sqlerr.RelationNotFound(rel)
+			}
+
+			if c, ok := tableMap[key]; ok {
 				a = append(a, Parameter{
 					Number: ref.ref.Number,
 					Column: &Column{

--- a/internal/endtoend/testdata/relation_does_not_exist/postgresql/query.sql
+++ b/internal/endtoend/testdata/relation_does_not_exist/postgresql/query.sql
@@ -1,0 +1,11 @@
+-- name: MissingSelect :many
+SELECT * FROM nonexisting_relation WHERE name = 'foo';
+
+-- name: MissingParam :many
+SELECT * FROM nonexisting_relation WHERE name = $1;
+
+-- name: MissingUpdate :exec
+UPDATE nonexisting_relation SET name = $1;
+
+-- name: MisingInsert :exec
+INSERT INTO nonexisting_relation (id) VALUES ($1);

--- a/internal/endtoend/testdata/relation_does_not_exist/postgresql/sqlc.json
+++ b/internal/endtoend/testdata/relation_does_not_exist/postgresql/sqlc.json
@@ -1,0 +1,11 @@
+{
+  "version": "1",
+  "packages": [
+    {
+      "path": "go",
+      "name": "querytest",
+      "schema": "query.sql",
+      "queries": "query.sql"
+    }
+  ]
+}

--- a/internal/endtoend/testdata/relation_does_not_exist/postgresql/stderr.txt
+++ b/internal/endtoend/testdata/relation_does_not_exist/postgresql/stderr.txt
@@ -1,0 +1,5 @@
+# package querytest
+query.sql:1:1: relation "nonexisting_relation" does not exist
+query.sql:5:1: relation "nonexisting_relation" does not exist
+query.sql:8:1: relation "nonexisting_relation" does not exist
+query.sql:11:1: relation "nonexisting_relation" does not exist

